### ISSUE TITLE
click sounds triggered multiple times instead of once

### ISSUE THREAD

**Issue by Simbaclaws** [MatiMenich/cordova-plugin-nativeClickSound#4]
Hey MatiMenich,

Apparently when I'm using your nativeclicksound plugin in ionic 3 some button click sounds will trigger multiple times, while others only trigger once or even not at all. Is this a issue with the plugin or is it ionic related?

I've tried both watching the buttons and a tags and also trying to trigger them manually. Even when triggering them manually I get multiple click sounds at once.
I'm going to continue checking out my code to be sure no click events are fired multiple times...

EDIT: It seems like event binding doesn't work properly outside of angular and causes random elements to fail registering as click handler. Whenever I find a tutorial on writing angular cordova plugins I'll rebuild this plugin into one that works for ionic.

EDIT2: It wasn't a angular issue or event binding issue but rather a nativeclick.js issue.

**Comment by Simbaclaws** [comment]
Could you please grant me access to push a new branch to this repository so I can create a pull request? I found the solution to my problem and want to contribute it to this repository.

Apparently the nativeclick.js has some code in it that causes weird bugs to appear. For example either not triggering the clicksound or triggering it multiple times at once.
I made a small adjustment to nativeclick.js that fixes these issues...

If you won't allow me access to add a branch and create a pull request, then here is the code that I changed:
```javascript
var exec = require('cordova/exec');

var nativeclick = {

  watch: function(classes){
    var clickyClasses = classes || [];

    for(var i = 0; i < clickyClasses.length; i++) {
      watchClass(clickyClasses[i]);
    }

    function watchClass(name){
      document.addEventListener('click', function(event) {
        if (event.target.classList.contains(name) || event.target.parentElement.classList.contains(name) || event.target.parentElement.parentElement.classList.contains(name) || event.target.parentElement.parentElement.parentElement.classList.contains(name)) {
          exec(null, errorCallback, "NativeClick", "click", []);
        }
      });
    }    
  },

  trigger: function () {
    exec(null, errorCallback, "NativeClick", "click", []);
  }
};

module.exports = nativeclick;

var errorCallback = function(error){
  console.error("Native Click Error:"+error);
};
```
All I did was change the watchClass function so that it moves up the dom tree to find the correct class inside the domtree, since event.target only targets the very inner element that is clicked through the DOM. Meaning that whenever there is for example a span inside a button with a class that should be making a click sound, that click sound won't appear because it's targeting the inner span instead of the button with the class. I've made it travel up the DOM tree 3 times to check any of the target and parents for the class that is passed.

Also window.event and event.srcElement seem to cause issues where clicks are registered multiple times causing a stuttering click sound to appear.

Plus using: .classList.contains(name) is more recommended and widely supported then using: .className.indexOf(name) != -1

**Comment by MatiMenich** [comment]
Sorry for the late response, I've been a little busy. I haven't had the time to check this out but it seems what you're saying is correct and I'll be happy to accept any pull request that you create.

You can create the pull request directly from the fork, it's not necessary for me to grant you rights to make a new branch. Checkout [this link](https://help.github.com/articles/creating-a-pull-request-from-a-fork/) 😉.

**Comment by Simbaclaws** [comment]
Alright I made the pull request have a look for yourself and do some tests if needed :+1: 

Also is it okay if I use your plugin as a base for another plugin I want to create? I want to trigger native push notification sounds when a push notification is recieved in the foreground of my app.
I could probably do it but I need a little head start.
Or might it be an idea to add that to this current plugin aswell?

Basically I found this java code somewhere:
```java
public void playNotificationSound() {
            Uri notification = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
            Ringtone r = RingtoneManager.getRingtone(mContext, notification);
            r.play();
    }
```

And this objective c code somewhere:

```objective c
AudioServicesPlaySystemSound(1007);
```

and this for ios vibration:
```objective c
AudioServicesPlaySystemSound(kSystemSoundID_Vibrate);
```

It could be a nice improvement by changing this plugin into a nativesound plugin instead of nativeclicksound plugin and adding all sorts of different native mobile sounds... Could be rather useful.
